lrpar/src/lib/parser.rs

```diff
@@ -573,13 +572,56 @@ pub(crate) mod test {
         assert_eq!(expected, pt.unwrap().pp(&grm, &input));
     }
 
+    // small_lexer and small_lex are our highly simplified version of lrlex (allowing us to avoid
+    // having to have lrlex as a dependency of lrpar). The format is the same as lrlex *except*:
+    //   * The initial "%%" isn't needed, and only "'" is valid as a rule name delimiter.
+    //   * "Unnamed" rules aren't allowed (e.g. you can't have a rule which discards whitespaces).
+    fn small_lexer(lexs: &str, ids_map: HashMap<String, u16>) -> Vec<(u16, Regex)> {
+        let mut rules = Vec::new();
+        for l in lexs.split("\n").map(|x| x.trim()).filter(|x| !x.is_empty()) {
+            println!("{}", l);
+            assert!(l.rfind("'") == Some(l.len() - 1));
+            let name_beg = l[..l.len() - 1].rfind("'").unwrap();
```

Comment:
what is a `beg`?

Reply:
"beginning" (maybe not the best name ever...)

Reply:
Maybe name it `name_start`?

Reply:
Will do.